c++/src/kj/compat/http.h

```diff
@@ -563,19 +565,32 @@ kj::Own<HttpClient> newHttpClient(HttpHeaderTable& responseHeaderTable, kj::Asyn
 // fail as well. If the destination server chooses to close the connection after a response,
 // subsequent requests will fail. If a response takes a long time, it blocks subsequent responses.
 // If a WebSocket is opened successfully, all subsequent requests fail.
+//
+// `entropySource` must be provided in order to use `openWebSocket`. If you don't need WebSockets,
+// `entropySource` can be omitted. The WebSocket protocol uses random values to avoid triggering
+// flaws (including security flaws) in certain HTTP proxy software. Specifically, entropy is used
+// to generate the `Sec-WebSocket-Key` header and to generate frame masks. If you know that there
+// are no broken or vulnerable proxies between you and the server, you can provide an dummy entropy
+// source that doesn't generate real entropy (e.g. returning the same value every time). Otherwise,
+// you must provide a cryptographically-random entropy source.
 
 kj::Own<HttpClient> newHttpClient(HttpService& service);
 kj::Own<HttpService> newHttpService(HttpClient& client);
 // Adapts an HttpClient to an HttpService and vice versa.
 
 kj::Own<WebSocket> newWebSocket(kj::Own<kj::AsyncIoStream> stream,
-                                kj::Maybe<WebSocket::MaskKeyGenerator&> maskKeyGenerator = nullptr);
+                                kj::Maybe<EntropySource&> maskEntropySource);
 // Create a new WebSocket on top of the given stream. It is assumed that the HTTP -> WebSocket
 // upgrade handshake has already occurred (or is not needed), and messages can immediately be
 // sent and received on the stream. Normally applications would not call this directly.
 //
-// `maskKeyGenerator` is optional, but if omitted, the WebSocket frames will not be masked. Refer
-// to RFC6455 to understand when masking is required.
+// `maskEntropySource` is used to generate cryptographically-random frame masks. If null, outgoing
+// frames will not be masked. Servers are not required to mask their outgoing frames, but clients
```

Comment:
nit: update comment to "Servers are required NOT to mask their outgoing frames ..."

[The RFC says](https://tools.ietf.org/html/rfc6455#section-5.1):

> A server MUST NOT mask any frames that it sends to the client.